File: Javascript/Array/E-766. Toeplitz Matrix.ts
function isToeplitzMatrix(matrix: number[][]): boolean {
  const m = matrix.length;
  const n = matrix[0].length;
  for (let i = 1; i < m; i++) {
    for (let j = 1; j < n; j++) {
      if (matrix[i][j] !== matrix[i - 1][j - 1]) return false;
    }
  }
  return true;
}

const testData: TestData[] = [
  {
    matrix: [
      [1, 2, 3, 4],
      [5, 1, 2, 3],
      [9, 5, 1, 2],
    ],
    ans: true,
  },
  {
    matrix: [
      [1, 2],
      [2, 2],
    ],
    ans: false,
  },
];

for (const data of testData) {
  console.log(data.ans === isToeplitzMatrix(data.matrix));
}

File: Array/E-766. Toeplitz Matrix/E-766. Toeplitz Matrix.py
class Solution:
    @classmethod
    def isToeplitzMatrix(self, matrix) -> bool:
        m, n = len(matrix), len(matrix[0])
        for i in range(1, m):
            for j in range(1, n):
                if matrix[i][j] != matrix[i-1][j-1]:
                    return False
        return True


test_data = [
    {
        "matrix": [
            [1, 2, 3, 4],
            [5, 1, 2, 3],
            [9, 5, 1, 2],
        ],
        "ans": True,
    },
    {
        "matrix": [
            [1, 2],
            [2, 2],
        ],
        "ans": False,
    },
]

for data in test_data:
    print(data["ans"] == Solution.isToeplitzMatrix(data["matrix"]))
